### PR TITLE
SearchBox fluent update

### PR DESCRIPTION
**Pull request by maxwellred**
#### Pull request checklist

- [ ] Addresses an existing issue: #7008 
- [x] Include a change request file using `$ npm run change`

#### Description of changes

This updates background, border, and icon color states per web fluent toolkit. The color changes are very subtle and can be best evaluated from the "Files Changes" tab. The noticeable visible changes have been accompanied below with a "(BEFORE)" screenshot.

#### Results
Rest
Screen Shot 2019-06-27 at 1 11 17 PM

Hover
Screen Shot 2019-06-27 at 1 11 08 PM

Focused + No Hover
Screen Shot 2019-06-27 at 1 11 33 PM

Populated + No Hover on Clear Button
Screen Shot 2019-06-27 at 1 18 21 PM

(BEFORE) Populated + No Hover on Clear Button
Screen Shot 2019-06-27 at 1 34 29 PM

Populated + Hover on Clear Button
Screen Shot 2019-06-27 at 1 11 55 PM

(BEFORE) Populated + Hover on Clear Button
Screen Shot 2019-06-27 at 1 34 37 PM

Populated + No Hover
Screen Shot 2019-06-27 at 1 12 50 PM

Underline (remove border radius)
Screen Shot 2019-06-27 at 1 12 24 PM

(BEFORE) Underline (remove border radius)
Screen Shot 2019-06-27 at 1 34 46 PM

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9611)

### CODE REVIEW

File: packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts [hunk 1]
```diff
@@ -1,15 +1,65 @@
...
+          ':hover ': {
+            borderColor: palette.neutralPrimary
+          },
+          ':active': {
+            borderColor: palette.themePrimary
+          }
+        }
+      },
+      !hasFocus &&
+        hasInput && {
```
Comment (Jahnp) [inline packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts]: I'm confused by this logic—what's happening here?

File: packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts [hunk 2]
```diff
@@ -1,15 +1,65 @@
...
+          selectors: {
+            '.ms-SearchBox-iconContainer': {
+              width: 0
```
Comment (Jahnp) [inline packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts]: What's the purpose of `width: 0`? Is this meant to hide the icon container?

File: packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts [hunk 3]
```diff
@@ -1,15 +1,65 @@
...
     root: [
       {
         borderRadius: effects.roundedCorner2,
-        borderColor: palette.neutralSecondaryAlt
+        borderColor: palette.neutralTertiary,
+        paddingTop: 0,
+        paddingBottom: 0,
+        selectors: {
+          ':hover ': {
+            borderColor: palette.neutralPrimary
```
Comment (dzearing) [inline packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts]: nit, but is it possible to snap to semantic color values here rather than palette colors? This would be a great forcing factor to ensure that TextField and SearchBox are identical. Could be a separate PR.

File: packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts (hunk 1 above)
Comment (maxwellred) [reply packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts]: Hi @Jahnp ! Thanks for commenting on my PR. So this logic specifically targets the scenario of an SearchBox having text but isn't currently focused on. Fabric 6 handles this by changing the border color AND allowing the magnifying glass to transition back into view. So per the toolkit interactive state called "Populated" in the Suite Header toolkit (scroll all the way to the bottom to "Secondary searchbox"), I'm removing the ability for the mag glass to transition back into view upon lost focus on a populated SearchBox.

Comment (Jahnp) [reply packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts]: Oh my gosh, my bad—in my haste I read `hasInput` as `hasFocus`, so when I commented I thought I was seeing this:
```typescript
!hasFocus &&
  hasFocus && {
  ...
```

Your explanation makes 100% sense and you can completely disregard this comment.

File: packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts (hunk 2 above)
Comment (maxwellred) [reply packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts]: Exactly! However I've since changed it to 4px to mimic Fabric 6's spec (to hide on ONLY focus)

File: packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts (hunk 3 above)
Comment (maxwellred) [reply packages/fluent-theme/src/fluent/styles/SearchBox.styles.ts]: Hello @dzearing ! That's a great point and I'll be making a PR for those changes shortly.

**Comment by maxwellred** [comment]
> If Ben says it's looking good on render I am OK from the point of view of the code.
> 
> @maxwellred Max Rediker (Webtellect LLC) Vendor will this be ported to the current master too?

Welcome back @Vitalius1 ! Yes, since this is now merged in i'll make the semanticColor updates in a different PR then port all the changes to master.
